src/Testable/Navigation.elm

```diff
@@ -0,0 +1,156 @@
+module Testable.Navigation exposing (..)
+
+import List.Extra
+import
+    -- This "unused" import is required for the native bindings
+    Native.Testable.Navigation
+import Navigation exposing (Location)
+import Regex exposing (HowMany(..), find, regex)
+
+
+type alias History =
+    ( Int, List Location )
```

Comment:
Ah, I was confused at first what the `Int` was for, but it makes sense from looking at the code. :)

1) I think the data structure here should be made private instead of being a type alias.

2) I think it would be ideal to use a list zipper instead of (Int, List).  There is wernerdegroot/listzipper, but it doesn't have nice functions for adding to the list.  Maybe we should have a ListZipper module in this project for now that either wraps wernerdegroot/listzipper or implements just what we need.   (In any case, doing that refactoring doesn't need to be done in this PR.)